compiler/z/codegen/UnaryEvaluator.cpp

```diff
@@ -520,9 +520,14 @@ OMR::Z::TreeEvaluator::l2aEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    node->setRegister(source);
 
 
-   //printf("Node %p is first child\n", firstChild); fflush(stdout);
-   //traceMsg(comp, "Node %p is first child\n", firstChild);
-   cg->decReferenceCount(firstChild);
+   if (shouldGenerateGuardedLoadsForCompressedLoad || firstChild->containsCompressionSequence())
```

Comment:
right, && is the correct one, I agree. Checked in the fix. 

I will run with -Xjit:paranoidOptCheck,enableParanoidRefCountChecks on p47. Do I need to rebuild with prod assume or will these options work on a regular build?